### DISCUSSION TITLE
Does openroad offer an API to transform rules file into an overall capacitance table?

### DISCUSSION THREAD

**Discussion by DTforwhat** [The-OpenROAD-Project/OpenROAD-flow-scripts#2559] [Q&A] [answered]
Is there a way to transform rules file into an captable which concludes total capacitance of each metal ?
or simply just read the rules file .
like this type 
M1
width(um)  space(um) Ctot(Ff/um)  Cc(Ff/um)    Carea(Ff/um) Cfrg(Ff/um)
0.070       0.052       0.1986       0.0723       0.0311       0.0115
0.070       0.065       0.1705       0.0509       0.0311       0.0143
0.070       0.200       0.1179       0.0115       0.0311       0.0319
0.070       0.335       0.1150       0.0030       0.0311       0.0388
0.070       0.470       0.1148       0.0009       0.0311       0.0409
0.070       0.605       0.1147       0.0002       0.0311       0.0416
0.070       0.740       0.1147       0.0001       0.0311       0.0417
0.070       0.875       0.1147       0.0000       0.0311       0.0418
0.210       0.052       0.2642       0.0727       0.0934       0.0127
0.210       0.065       0.2358       0.0512       0.0934       0.0155
0.210       0.200       0.1828       0.0115       0.0934       0.0331
0.210       0.335       0.1798       0.0030       0.0934       0.0401
0.210       0.470       0.1796       0.0009       0.0934       0.0422
0.210       0.605       0.1796       0.0002       0.0934       0.0428
0.210       0.740       0.1796       0.0001       0.0934       0.0430
0.210       0.875       0.1796       0.0000       0.0934       0.0431
1.000       0.052       0.6285       0.0727       0.4449       0.0191
1.000       0.065       0.6002       0.0512       0.4449       0.0219
1.000       0.200       0.5471       0.0115       0.4449       0.0396
1.000       0.335       0.5442       0.0030       0.4449       0.0465
1.000       0.470       0.5440       0.0009       0.4449       0.0487
1.000       0.605       0.5440       0.0002       0.4449       0.0493
1.000       0.740       0.5440       0.0001       0.4449       0.0495
1.000       0.875       0.5440       0.0000       0.4449       0.0495
9.000       0.052       4.3181       0.0727       4.0043       0.0842
9.000       0.065       4.2898       0.0512       4.0043       0.0870
9.000       0.200       4.2367       0.0115       4.0043       0.1047
9.000       0.335       4.2338       0.0030       4.0043       0.1116
9.000       0.470       4.2336       0.0009       4.0043       0.1138
9.000       0.605       4.2335       0.0002       4.0043       0.1144
9.000       0.740       4.2335       0.0001       4.0043       0.1146
9.000       0.875       4.2335       0.0000       4.0043       0.1146

**Comment by maliberty** [comment]
What does "total capacitance of each metal" mean?  Capacitance depends on the geometry of the wires (widths, spacings, layers)

**Reply by DTforwhat** [reply]
thx for replying
I mean a parasitic capacitance table classified according to different wire width and space for each metal layer

**Reply by maliberty** [reply, marked as answer]
No though even such a table would have to make assumptions about other layers and symmetry on a single layer.  You could pull Ctot from the above as a very basic model.

**Reply by DTforwhat** [reply]
thank u so much for replying so quickly
